WooCommerce/Classes/ViewRelated/Products/Edit Product/Edit Price/Product+PriceSettingsViewModels.swift

```diff
@@ -11,43 +11,56 @@ extension Product {
 
     static func createRegularPriceViewModel(regularPrice: String?,
                                             using currencySettings: CurrencySettings,
-                                       onInputChange: @escaping (_ input: String?) -> Void) -> UnitInputViewModel {
+                                            onInputChange: @escaping (_ input: String?) -> Void) -> UnitInputViewModel {
         let title = NSLocalizedString("Price", comment: "Title of the cell in Product Price Settings > Price")
 
         let currencyFormatter = CurrencyFormatter()
         let currencyCode = CurrencySettings.shared.currencyCode
         let unit = CurrencySettings.shared.symbol(from: currencyCode)
-        var value = currencyFormatter.formatAmount(regularPrice ?? "", with: unit) ?? ""
-        value = value
-            .replacingOccurrences(of: unit, with: "")
-            .replacingOccurrences(of: regexThousandSeparators, with: "$1", options: .regularExpression)
+        let decimalValue = regularPrice.map { currencyFormatter.convertToDecimal(from: $0) ?? 0 }
+        let value = decimalValue.map { "\($0)" } ?? ""
```

Comment:
@pmusolino Would you mind double-checking this? It doesn't look like it'll cause a regression of what https://github.com/woocommerce/woocommerce-ios/pull/1802 fixed but I'm not sure.

Reply:
If this is good, we can remove the `regexThousandSeparators` constant.

Reply:
I confirm that there are no regressions, and the tests that I wrote on the previous PR are green 👍 Well done @jaclync! 😉

Reply:
Thank you, Paolo! 🙏